db/helpers/botDataUtils.js

```diff
@@ -190,98 +191,95 @@ function updateValues(seq, instance) {
-          Object.keys(context[syncBot]).forEach((botValueName) => {
-            syncValue = isJson(context[syncBot][botValueName]) ?
-              serialize(context[syncBot][botValueName]) :
-              context[syncBot][botValueName];
-            const replaceBlocks = syncValue.match(/\${(.*?)}/g);
+      return [];
+    })
+    .then((botJSON) => {
+      // Create lookups for bots by name or id
+      botJSON.forEach((bot) => {
+        botsIds[bot.dataValues.id] = bot.dataValues.name;
+        botNames[bot.dataValues.name] = bot.dataValues.id;
+      });
+    })
+    .then(() => {
+      // Determine if a sync needs to be performed
+      if (room && ('settings' in room) && ('sharedContext' in room.settings)) {
+        const settings = room.settings.sharedContext;
+        const context = settings[botsIds[instance.botId]];
+        const promises = [];
+        if (context) {
+          // Create a different sync for each bot listed
+          Object.keys(context).forEach((syncBot) => {
+            let whereConst = {};
+            let syncValue = '';
+            Object.keys(context[syncBot]).forEach((botValueName) => {
+              syncValue = isJson(context[syncBot][botValueName]) ?
+                serialize(context[syncBot][botValueName]) :
+                context[syncBot][botValueName];
+              const replaceBlocks = syncValue.match(/\${(.*?)}/g);
 
-            // Replace logic blocks wtih vlaues
-            replaceBlocks.forEach((replaceBlock) => {
-              const replaceField = replaceBlock.replace('${', '')
-                .replace('}', '').split('.');
-              if (instance.dataValues.name === replaceField[ONE]) {
-                syncValue = replaceValue(
-                  syncValue,
-                  replaceBlock,
-                  instance.dataValues
-                );
-                whereConst = {
-                  where:
-                  {
-                    name: botValueName,
-                    botId: botNames[syncBot],
-                    roomId: instance.roomId,
-                  },
-                };
-              }
+              // Replace logic blocks wtih vlaues
+              replaceBlocks.forEach((replaceBlock) => {
+                const replaceField = replaceBlock.replace('${', '')
+                  .replace('}', '').split('.');
+                if (instance.dataValues.name === replaceField[ONE]) {
+                  syncValue = replaceValue(
+                    syncValue,
+                    replaceBlock,
+                    instance.dataValues
+                  );
+                  whereConst = {
+                    where: {
+                      name: botValueName,
+                      botId: botNames[syncBot],
+                      roomId: instance.roomId,
+                    },
+                  };
+                }
+              });
             });
+
+            // Find bot data to update
+            promises.push(BotData.findOne(whereConst));
```

Comment:
why did you remove the catch and logging?

Reply:
was not here, already added - it was removed by mistake

Reply:
cool :)